Fix Instance Handling of HelpWindow and ValueTable

FabianRolfMatthiasNoll opened the pull request:
Making the HelpWindow just invisible and visible made problems in combination with the valueTables and other HelpWindow instances

The same procedure of the HelpWindow is now used on ValueTable which makes handling it way easier and safer. Also, much of the "complicated" code for checking the states got removed, and the ValueTable is now dynamic and not hardcoded to 3 instances anymore

FabianRolfMatthiasNoll committed:
Change Behavior of HelpWindow instance handling

Making the HelpWindow just invisible and visible made problems in combination with the valueTables and other HelpWindow instances

FabianRolfMatthiasNoll committed:
Change Behavior of ValueTable Instance handling

The same procedure of the HelpWindow is now used on ValueTable which makes handling it way easier and safer. Also, much of "complicated" code for checking the states got removed, and the ValueTable is now dynamic and not hardcoded to 3 instances anymore

kantholz44mm commented on src/GUI/HelpWindow.java:
```diff
@@ -29,32 +28,33 @@ private HelpWindow() {
-            instance = new HelpWindow();
-        }
-        return instance;
+    private void setWindowSizeAndPosition(){
```
just from the name, I would assume the function sets this to a position/size given by a parameter, maybe rename this to something like "centerWindow" or something

FabianRolfMatthiasNoll replied:
I disagree because it doesnt just center the window it also sets the size and i dont think implies a parameter

kantholz44mm reviewed (approved):
one minor nitpick, change if you agree, otherwise good to go

FabianRolfMatthiasNoll committed:
Fix Graph not deletable

The Graph wasn't able to be deleted when a graph with a lower index was deleted because the indexes weren't correct anymore

FabianRolfMatthiasNoll committed:
make dirty fix to stable fix

kantholz44mm commented on src/GUI/GraphPanel.java:
```diff
@@ -94,8 +92,8 @@ private Vector2D findNearestGraphPoint(Vector2D screenPosition) {
 
-        for (ParametricFunction function : functions) {
-            if(function == null) continue;
+        for (UUID uuid : functions.keySet()) {
```
why not just loop through functions.values() if we only need the function, and not the ID?

FabianRolfMatthiasNoll replied:
Done

kantholz44mm commented on src/GUI/GraphPanel.java:
```diff
@@ -310,8 +308,8 @@ private void drawFunctions(Graphics2D g2d) {
 
-        for (ParametricFunction parametricFunction : functions) {
-            if(parametricFunction == null) continue;
+        for (UUID uuid : functions.keySet()) {
```
again, we can just loop through functions.values()

kantholz44mm commented on src/GUI/GraphPanel.java:
```diff
@@ -354,8 +352,8 @@ private void drawInformationWindows(Graphics2D g2d) {
         int index = 0;
-        for (Object function : functions) {
-            if(function == null) continue;
+        for (UUID uuid : functions.keySet()) {
```
loop through functions.values()

kantholz44mm reviewed (commented):
I think you can loop through the functions directly instead of doing functions.get() everytime. Otherwise looks solid

FabianRolfMatthiasNoll committed:
Loop through Iterations directly
